File: imob-erp-frontend/src/types/property.ts
export type PropertyType = "CASA" | "APARTAMENTO" | "COMERCIAL" | "TERRENO";
export type PropertyStatus = "DISPONIVEL" | "RESERVADO" | "VENDIDO" | "ALUGADO";
export type PropertyPurpose = "VENDA" | "ALUGUEL" | "AMBOS";

export interface Property {
  id: string;
  type: PropertyType;
  title: string;
  description?: string;
  address: string;
  neighborhood?: string;
  city: string;
  price: number;
  area?: number;
  bedrooms?: number;
  bathrooms?: number;
  parkingSpots?: number;
  status: PropertyStatus;
  purpose: PropertyPurpose;
  photos: string[];
}

export type PropertyRequest = Omit<Property, "id" | "status" | "photos">;

export interface PropertyFilters {
  type?: PropertyType;
  status?: PropertyStatus;
  neighborhood?: string;
  minPrice?: number;
  maxPrice?: number;
}

File: imob-erp-backend/src/main/java/com/imobcrm/property/domain/Property.java
package com.imobcrm.property.domain;

import com.imobcrm.property.domain.enums.PropertyPurpose;
import com.imobcrm.property.domain.enums.PropertyStatus;
import com.imobcrm.property.domain.enums.PropertyType;
import com.imobcrm.shared.audit.AuditEntity;
import jakarta.persistence.Column;
import jakarta.persistence.ElementCollection;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Id;
import jakarta.persistence.OrderColumn;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "properties")
public class Property extends AuditEntity {

    @Id
    private UUID id;

    @Column(name = "tenant_id", nullable = false)
    private UUID tenantId;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PropertyType type;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "text")
    private String description;

    @Column(nullable = false)
    private String address;

    private String neighborhood;

    @Column(nullable = false)
    private String city;

    @Column(nullable = false, precision = 14, scale = 2)
    private BigDecimal price;

    @Column(precision = 10, scale = 2)
    private BigDecimal area;

    private Integer bedrooms;

    private Integer bathrooms;

    @Column(name = "parking_spots")
    private Integer parkingSpots;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PropertyStatus status;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PropertyPurpose purpose;

    @ElementCollection
    @OrderColumn(name = "photo_order")
    @Builder.Default
    private List<String> photos = new ArrayList<>();

    @Column(nullable = false)
    @Builder.Default
    private boolean active = true;
}
